docs/src/catalyst_applications/jump_simulation_performance.md

```diff
@@ -0,0 +1,163 @@
+
+Here are some example reactions for the different types of jumps:
+```@example jump_simulation_performance_1
+# `MassActionJump`s
+@reaction 1.0, X --> Y
+@reaction k, 2X + Y --> X2Y
+@reaction k1*k2+k3, 0 --> X
+
+# `ConstantRateJump`s
+@reaction k*log(X), Y --> X
+@reaction mm(X,v,K), 0 --> X
+
+# `VariableRateJump`s
+@reaction k*(1+sin(t)), 0 --> X
+@reaction X/t, X + Y --> XY
+nothing # hide
+```
+Throughout a simulation, `VariableRateJump`s' rates require updating more frequently than `ConstantRateJump`s' rates, which in turn requires updating more frequently than `MassActionJump`s' rates. Since the performance of jump simulations is heavily affected by how frequently jump rates are computed, keeping the number of `ConstantRateJump`s and `VariableRateJump`s as small as possible is advantageous. Primarily, there exist two common cases where models are written in a way so that sub-optimal jump types are generated, both of which we describe below.
+
+#### Unnecessarily putting species in rates rather than in reactions
+Sometimes, a rate has been made dependent on a species, where that species instead could have been made part of the actual reaction. Consider a reaction where an enzyme ($E$) catalyses the phosphorylation of a protein ($X$) to phosphorylated form ($Xᵖ$). It can be written in two different forms:
+```@example jump_simulation_performance_1
+r1 = @reaction k*E, X --> Xᵖ 
+r2 = @reaction k, X + E --> Xᵖ + E
+nothing # hide
+```
+These two reactions will generate identical simulations (this holds for ODE, SDE, and Jump simulations). However, while `r1` will generate a `ConstantRateJump`, `r2` will generate a `MassActionJump`. Hence, if the `r2` form is used, jump simulation performance is, at no cost, improved. Since the two forms are otherwise identical, it is always preferable to, whenever possible, put species in the reactions, rather than the rates.
+
+#### Using piecewise constant, time-dependant, function instead of callbacks.
+Let us consider a reaction with some input that depends on time. We assume that the input is initially $0$, but after some critical time $t=5.0$ it is increased to $1$. This can be implemented through a step function:
+```@example jump_simulation_performance_1
+input(t) = t > 5.0
+r = @reaction input(t), 0 --> X
+nothing # hide
+```
+Here, the production of species $X$ is switched on at the time $t=5.0$. This reaction (which rate depends on `t`) will generate a `VariableRateJump`, which is bad for jump simulation performance. However, since the rate is piecewise constant, it can instead be implemented by setting it to a constant parameter $i$, and then use a [*callback*](@ref advanced_simulations_callbacks) to update it at the critical times. This will again generate an equivalent model, but with the reaction encoded as a `MassActionJump` (rather than a `VariableRateJump`).
```

Comment:
```suggestion
Here, the production of species $X$ is switched on at the time $t=5.0$. This reaction (for which the rate depends on `t`) will generate a `VariableRateJump`, which is the least performant jump type in simulations, and can not be used with the standard SSAs of JumpProcesses. However, since the rate is piecewise constant, it can alternatively be implemented by setting it to a constant parameter $i$, and then using a [*discrete callback*](@ref advanced_simulations_callbacks) to update it at the switching times. This will again generate an equivalent model, but with the reaction encoded as a `MassActionJump` (rather than a `VariableRateJump`). Generally such explicit time-discontinuities should be encoded via discrete callbacks instead of as `VariableRateJump`s if possible as simulation methods for the latter typically assume the system's propensities evolve continuously in-between jumps.
```
Do you want to show how to do this?

Should we show how to actually simulate a `VariableRateJump` model? (The interface isn't perfect, but it can be done...)